code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaSession.java

```diff
@@ -272,15 +274,24 @@ private void trySendHit() {
                             Log.debug(
                                     MediaInternalConstants.EXTENSION_LOG_TAG,
                                     LOG_TAG,
-                                    "trySendHit - (%s) Finished http connection",
+                                    "trySendHit - (%s) Finished http connection.",
                                     hitEventType);
 
                             synchronized (mutex) {
                                 boolean shouldRetry = false;
 
-                                if (hitIsSessionStart
+                                if (deviceOffline) {
+                                    // `trySendHit` verifies network connectivity before sending a
```

Comment:
can you also add a reference to isReadyToSendHit where this verification happens

Reply:
Do we want me to update the comment with this function reference?